src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -173,17 +167,34 @@ private static RedirectionItem[] LoadRedirectionModel(ErrorBuilder errors, strin
                     // Rebase source_path based on redirection definition file path
                     var basedir = Path.GetDirectoryName(fullPath) ?? "";
 
-                    return (
-                        from item in redirections.Concat(renames)
-                        let sourcePath = Path.GetRelativePath(docsetPath, Path.Combine(basedir, item.SourcePath))
-                        where !sourcePath.StartsWith(".")
-                        select new RedirectionItem
+                    List<RedirectionItem> results = new List<RedirectionItem>();
+                    foreach (var item in redirections.Concat(renames))
+                    {
+                        var sourcePath = Path.GetRelativePath(docsetPath, Path.Combine(basedir, item.SourcePath));
+                        if (!string.IsNullOrEmpty(item.RedirectUrl.Value) && !item.SourcePath.IsDefault)
+                        {
+                            results.Add(new RedirectionItem
+                            {
+                                SourcePath = new PathString(sourcePath),
+                                Monikers = item.Monikers,
+                                RedirectUrl = item.RedirectUrl,
+                                RedirectDocumentId = item.RedirectDocumentId,
+                            });
+                        }
+
+                        if (item.SourcePath.IsDefault)
```

Comment:
put these two `if` check ahead, `continue` within them.
Then we can skip the previous duplicated `if` check.